lib/resolveurl/lib/net.py

```diff
@@ -310,8 +310,6 @@ def _fetch(self, url, form_data={}, headers={}, compression=True, jdata=False):
         if form_data:
             if jdata:
                 form_data = json.dumps(form_data)
-            elif isinstance(form_data, six.string_types):
```

Comment:
It is intentionally there and not useless as you say ;-)
The formdata is left alone if it is already a string and urlencoded it is a dictionary
Yes, it could have been done as
```
elif not isinstance(form_data, six.string_types):
    form_data = urllib_parse.urlencode(form_data, True)
```
and then remove the else statement completely.
However if your proposed change is merged, then raw post fucntionality would be broken

Reply:
ok then you can close here
thanks for the explanation